### PR TITLE
Task/Update deployment logic

### PR DESCRIPTION
**Pull request by ajrice6713**

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
+    types:
+      - published
+
 
 jobs:
-  build:
-    runs-on: windows-latest
+  publish:
+    name: Publish
+    if: ${{ !github.event.release.prerelease && github.event.release.target_commitish == 'main' }}
```
Comment (hamermike) [inline .github/workflows/deploy.yaml:11]:
```suggestion
    if: ${{ !github.event.release.prerelease && github.event.release.target_commitish == 'master' }}
```

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
-      shell: cmd
-      env:
-        NUGET_KEY: ${{ secrets.NUGET_KEY }}
+      - uses: actions/checkout@v2
+      - name: Set release version
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/v}" >> $GITHUB_ENV
+      - uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+      - name: Run integration tests
```
Comment (hamermike) [inline .github/workflows/deploy.yaml:23]:
```suggestion
      - name: Run unit tests
```

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
+        env:
+          BW_ACCOUNT_ID: ${{ secrets.BW_ACCOUNT_ID }}
+          BW_USERNAME: ${{ secrets.BW_USERNAME }}
+          BW_PASSWORD: ${{ secrets.BW_PASSWORD }}
+          BW_VOICE_APPLICATION_ID: ${{ secrets.BW_VOICE_APPLICATION_ID }}
+          BW_MESSAGING_APPLICATION_ID: ${{ secrets.BW_MESSAGING_APPLICATION_ID }}
+          BW_NUMBER: ${{ secrets.BW_NUMBER }}
+          USER_NUMBER: ${{ secrets.USER_NUMBER }}
+          BASE_CALLBACK_URL: ${{ secrets.BASE_CALLBACK_URL }}
```
Comment (hamermike) [inline .github/workflows/deploy.yaml:32]:
```suggestion
```

File: .github/workflows/release-draft.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    name: deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # Drafts your next Release notes as Pull Requests are merged
+      - uses: release-drafter/release-drafter@v5
+        with:
+          config-name: draft_release.yml
+        env:
+          GITHUB_TOKEN: ${{ secrets.BW_GITHUB_TOKEN }}
```
Comment (hamermike) [inline .github/workflows/release-draft.yml:19]: I don't know if this is actually required and would suggest we attempt without it for the moment.
```suggestion
        env:
          GITHUB_TOKEN: ${{ secrets.BW_GITHUB_TOKEN }}
```

**Review by hamermike — CHANGES_REQUESTED** [review]

**Review by hamermike — APPROVED** [review]

**Review by ckoegel — APPROVED** [review]

**Pull request merged**